iothub_client/src/uamqp_messaging.c

```diff
@@ -24,6 +24,10 @@
 
 #define MESSAGE_ID_MAX_SIZE 128
 
+#define AMQP_DIAGNOSTIC_ID_KEY "Diagnostic-Id"
```

Comment:
Please make sure the name clearly links to E2E diagnostics.

Reply:
What I mean is that the name should have E2E diagnostics somehow in it

Reply:
please see my comment below.